R/chat.R

```diff
@@ -49,50 +55,79 @@ NULL
 #'   Briefly explain your work.
 #' ")
 new_chat_openai <- function(system_prompt = NULL,
+                            messages = NULL,
                             base_url = "https://api.openai.com/v1",
                             api_key = openai_key(),
-                            model = "gpt-4o-mini",
+                            model = NA,
                             echo = FALSE) {
   check_string(system_prompt, allow_null = TRUE)
   check_string(base_url)
   check_string(api_key)
-  check_string(model)
+  check_string(model, allow_null = TRUE, allow_na = TRUE)
   check_bool(echo)
 
-  system_prompt <- system_prompt %||%
-    "You are a helpful assistant from New Zealand who is an experienced R programmer"
+  if (is.na(model)) {
+    model <- "gpt-4o-mini"
+  }
+
+  # TODO: Sanity check individual messages?
+  if (!is.null(messages)) {
```

Comment:
A bit prospective maybe, but I'd suggest pulling this out into `check_openai_conversation()`